src/main/java/org/spongepowered/common/data/manipulator/immutable/common/AbstractImmutableBoundedComparableData.java

```diff
@@ -57,22 +56,42 @@
     protected final Comparator<T> comparator;
     protected final T lowerBound;
     protected final T upperBound;
+    protected final T defaultValue;
+    protected final ImmutableBoundedValue<T> immutableBoundedValue;
 
     protected AbstractImmutableBoundedComparableData(Class<I> immutableClass, T value,
-                                                  Key<? extends BaseValue<T>> usedKey,
-                                                  Comparator<T> comparator, Class<? extends M> mutableClass, T lowerBound, T upperBound) {
+                                                     Key<? extends BaseValue<T>> usedKey,
+                                                     Comparator<T> comparator, Class<? extends M> mutableClass,
+                                                     T lowerBound, T upperBound, T defaultValue) {
         super(immutableClass, value, usedKey);
         this.comparator = comparator;
         checkArgument(!Modifier.isAbstract(mutableClass.getModifiers()), "The immutable class cannot be abstract!");
         checkArgument(!Modifier.isInterface(mutableClass.getModifiers()), "The immutable class cannot be an interface!");
+        checkArgument(comparator.compare(defaultValue, lowerBound) >= 0 && comparator.compare(defaultValue, upperBound) <= 0 );
         this.mutableClass = mutableClass;
         this.lowerBound = lowerBound;
         this.upperBound = upperBound;
+        this.defaultValue = defaultValue;
+        if (this.comparator.compare(this.upperBound, this.lowerBound) <= ImmutableDataCachingUtil.CACHE_LIMIT_FOR_INDIVIDUAL_TYPE) {
```

Comment:
Will this work for float/double values?

20F is within this bound, but has many more than 100 potential values.